src/docs/development/add-to-app/ios/add-flutter-screen.md

```diff
@@ -0,0 +1,487 @@
+    [self presentViewController:flutterViewController animated:false completion:nil];
+}
+@end
+```
+
+Or, using Swift:
+
+`ViewController.swift`:
+
+```swift
+import UIKit
+import Flutter
+
+class ViewController: UIViewController {
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    let button = UIButton(type:UIButton.ButtonType.custom)
+    button.addTarget(self, action: #selector(handleButtonAction), for: .touchUpInside)
+    button.setTitle("Press me", for: UIControl.State.normal)
+    button.frame = CGRect(x: 80.0, y: 210.0, width: 160.0, height: 40.0)
+    button.backgroundColor = UIColor.blue
+    self.view.addSubview(button)
+  }
+
+  @objc func handleButtonAction() {
+    let flutterEngine = (UIApplication.shared.delegate as? AppDelegate)?.flutterEngine;
+    let flutterViewController = FlutterViewController(engine: flutterEngine, nibName: nil, bundle: nil)!;
+    self.present(flutterViewController, animated: false, completion: nil)
+  }
+}
+```
+
+You should now be able to build and launch MyApp on the Simulator or on a device.
+Pressing the button should bring up a full-screen Flutter view with the standard
+Flutter Demo counting app. You can use routes to show different widgets at different
+places in your app, as described in the Android section above. To set the route, call
+
+- Objective-C:
+
+```objective-c
+[flutterViewController setInitialRoute:@"route1"];
+```
+
+- Swift:
+
+```swift
+flutterViewController.setInitialRoute("route1")
+```
+
+immediately after construction of the `FlutterViewController` (and before presenting
+it).
+
+You can have the Flutter app dismiss itself by calling `SystemNavigator.pop()`
+in the Dart code.
+
+## Building and running your app
+
+You build and run MyApp using Xcode in exactly the same way that you did before you
+added the Flutter module dependency. The same goes for editing, debugging, and
+profiling your iOS code.
+
+## Hot restart/reload and debugging Dart code
+
+Connect a device or launch a Simulator. Then make Flutter CLI tooling listen
+for your app to come up:
+
+```bash
+$ cd some/path/my_flutter
+$ flutter attach
+Waiting for a connection from Flutter on iPhone X...
+```
+
+Launch `MyApp` in debug mode from Xcode. Navigate to an area of the app that uses
+Flutter. Then turn back to the terminal, and you should see output similar to the
+following:
+
+```text
+Done.
+Syncing files to device iPhone X...                          4.7s
+
+🔥  To hot reload changes while running, press "r". To hot restart (and rebuild state), press "R".
+An Observatory debugger and profiler on iPhone X is available at: http://127.0.0.1:54741/
+For a more detailed help message, press "h". To quit, press "q".
+```
+
+You can now edit the Dart code in `my_flutter`, and the changes can be hot
+reloaded by pressing `r` in the terminal. You can also paste the URL above into
+your browser to use the Dart Observatory for setting breakpoints, analyzing
+memory retention and other debugging tasks.
```

Comment:
OK, can this be accomplished using Dart DevTools? We've been asked NOT to highlight/use Observatory in our docs if it can be done with DevTools.

@jacob314

Reply:
I'm not sure of the current status of DevTools.